osu.Game.Rulesets.Catch/UI/CatcherTrailDisplay.cs

```diff
@@ -0,0 +1,138 @@
+    /// </summary>
+    public class CatcherTrailDisplay : CompositeDrawable
+    {
+        private readonly Catcher catcher;
+
+        private readonly Container<CatcherTrailSprite> dashTrails;
+        private readonly Container<CatcherTrailSprite> hyperDashTrails;
+        private readonly Container<CatcherTrailSprite> endGlowSprites;
+
+        private Color4 hyperDashTrailsColour;
+
+        public Color4 HyperDashTrailsColour
+        {
+            get => hyperDashTrailsColour;
+            set
+            {
+                if (hyperDashTrailsColour == value)
+                    return;
+
+                hyperDashTrailsColour = value;
+                hyperDashTrails.FadeColour(hyperDashTrailsColour, Catcher.HYPER_DASH_TRANSITION_DURATION, Easing.OutQuint);
+            }
+        }
+
+        private Color4 endGlowSpritesColour;
+
+        public Color4 EndGlowSpritesColour
+        {
+            get => endGlowSpritesColour;
+            set
+            {
+                if (endGlowSpritesColour == value)
+                    return;
+
+                endGlowSpritesColour = value;
+                endGlowSprites.FadeColour(endGlowSpritesColour, Catcher.HYPER_DASH_TRANSITION_DURATION, Easing.OutQuint);
+            }
+        }
+
+        private bool trail;
+
+        /// <summary>
+        /// Whether to start displaying trails following the catcher.
+        /// </summary>
+        public bool DisplayTrail
+        {
+            get => trail;
+            set
+            {
+                if (trail == value)
+                    return;
+
+                trail = value;
+
+                if (trail)
+                    displayTrail();
+            }
+        }
+
+        public CatcherTrailDisplay([NotNull] Catcher catcher)
+        {
+            this.catcher = catcher ?? throw new ArgumentNullException(nameof(catcher));
+
+            RelativeSizeAxes = Axes.Both;
+
+            InternalChildren = new[]
+            {
+                dashTrails = new Container<CatcherTrailSprite> { RelativeSizeAxes = Axes.Both },
+                hyperDashTrails = new Container<CatcherTrailSprite> { RelativeSizeAxes = Axes.Both, Colour = Catcher.DEFAULT_HYPER_DASH_COLOUR },
+                endGlowSprites = new Container<CatcherTrailSprite> { RelativeSizeAxes = Axes.Both, Colour = Catcher.DEFAULT_HYPER_DASH_COLOUR },
+            };
+        }
+
+        /// <summary>
+        /// Displays a single end-glow catcher sprite.
+        /// </summary>
+        public void DisplayEndGlow()
+        {
+            var endGlow = createTrailSprite(endGlowSprites);
+
+            endGlow.MoveToOffset(new Vector2(0, -10), 1200, Easing.In);
+            endGlow.ScaleTo(endGlow.Scale * 0.95f).ScaleTo(endGlow.Scale * 1.2f, 1200, Easing.In);
+            endGlow.FadeOut(1200);
+            endGlow.Expire(true);
+        }
+
+        private void displayTrail()
+        {
+            if (!catcher.Dashing && !catcher.HyperDashing)
```

Comment:
I find it super weird that `Catcher` is responsible for starting the trail and this new class is responsible for stopping it...